cpp/oneapi/dal/backend/common.hpp

```diff
@@ -59,6 +59,36 @@ inline constexpr Integer up_multiple(Integer x, Integer multiple) {
     return y + z;
 }
 
+/// Finds the largest power of 2 number not larger than `x`.
+/// Return `x`, if `x` is already power of 2
+/// Example: down_pow2(10) == 8
+/// Example: down_pow2(16) == 16
+template <typename Integer>
+inline constexpr Integer down_pow2(Integer x) {
```

Comment:
Do these functions really common for whole backend, or it just some math primitives?

Reply:
I think this is common enough, for example it's used to determine appropriate local range of DPC++ kernel.